java/src/org/openqa/selenium/grid/node/remote/RemoteNode.java

```diff
@@ -83,13 +86,15 @@ public RemoteNode(
       NodeId id,
       URI externalUri,
       Secret registrationSecret,
+      Duration sessionTimeout,
       Collection<Capabilities> capabilities) {
-    super(tracer, id, externalUri, registrationSecret);
+    super(tracer, id, externalUri, registrationSecret, sessionTimeout);
     this.externalUri = Require.nonNull("External URI", externalUri);
     this.capabilities = ImmutableSet.copyOf(capabilities);
 
-    this.client =
-        Require.nonNull("HTTP client factory", clientFactory).createClient(fromUri(externalUri));
+    ClientConfig clientConfig =
+        defaultConfig().readTimeout(this.getSessionTimeout()).baseUrl(fromUri(externalUri));
```

Comment:
@diemol, do you think connectionTimeout() also need here?

Reply:
I don't think so. Connection timeout is used for the initial connection, and in theory all components are in easily reachable network(s).

Reply:
Sure, so we set `readTimeout` only.